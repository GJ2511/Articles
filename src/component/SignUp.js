import React from 'react';

function SignUp() {
    return (
        <div className="row">
            <div className="col-sm-9 col-md-7 col-lg-5 mx-auto">
                <div className="card card-signUp my-5 bg-light ">
                    <div className="card-body">
                        <h5 className="card-title text-center">Sign Up</h5>
                        <form>
                            <div className="form-group">
                                <input
                                    type="text"
                                    className="form-control rounded-pill"
                                    id="username"
                                    aria-describedby="usernameHelp"
                                    placeholder="User Name"
                                    autoFocus
                                />
                            </div>
                            <div className="form-group">
                                <input
                                    type="email"
                                    className="form-control rounded-pill"
                                    id="email"
                                    aria-describedby="emailHelp"
                                    placeholder="Email address"
                                />
                            </div>
                            <div className="form-group">
                                <input
                                    type="password"
                                    className="form-control rounded-pill"
                                    id="password"
                                    placeholder="Password "
                                />
                            </div>
                            <button type="submit" className="btn btn-primary btn-lg btn-block rounded-pill">
                                Sign Me Up
                            </button>
                        </form>
                    </div>
                </div>
            </div>
        </div>
    );
}

export default SignUp;
