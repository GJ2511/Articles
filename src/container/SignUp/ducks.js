import { takeLatest, put, call } from 'redux-saga/effects';

import AuthService from '../../services/authService';
import { setSignInSuccess } from '../../ducks';

const PREFIX = 'SIGN_UP';
const SIGN_UP_REQUESTED = `${PREFIX}//SIGN_UP_REQUESTED`;
const SIGN_UP_SUCCESS = `${PREFIX}//SIGN_UP_SUCCESS`;
const SIGN_UP_FAILED = `${PREFIX}//SIGN_UP_FAILED`;

const initialState = {
    isSubmitting: false,
    error: {},
};

const signUpReducer = (state = initialState, action = {}) => {
    const { type, payload } = action;

    switch (type) {
        case SIGN_UP_REQUESTED:
            return {
                ...state,
                isSubmitting: true,
                error: {},
            };
        case SIGN_UP_FAILED:
            return {
                ...state,
                isSubmitting: false,
                error: payload,
            };
        default:
            return state;
    }
};

export const signUpRequested = (values) => ({
    type: SIGN_UP_REQUESTED,
    payload: values,
});

function* signUp({ payload: { username, email, password } }) {
    try {
        const response = yield call([AuthService, 'signup'], { username, email, password });

        if (response.errors) {
            yield put({ type: SIGN_UP_FAILED, payload: response.errors });
        } else {
            yield put({ type: SIGN_UP_SUCCESS });
            yield put(setSignInSuccess(response.user));
        }
    } catch (error) {
        yield put({
            type: SIGN_UP_FAILED,
            payload: {
                Error: 'Something went wrong',
            },
        });
    }
}

export function* signUpSaga() {
    yield takeLatest(SIGN_UP_REQUESTED, signUp);
}

export const actions = {
    signUpRequested,
};

export default signUpReducer;
