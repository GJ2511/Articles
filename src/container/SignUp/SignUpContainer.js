import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Formik } from 'formik';

import SignUp from '../../component/SignUp';
import { signUpRequested } from './ducks';

const initialValues = {
    username: '',
    email: '',
    password: '',
};

class SignUpContainer extends Component {
    handleValidation = (values) => {
        const errors = {};

        if (!values.email) {
            errors.email = 'Required';
        } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(values.email)) {
            errors.email = 'Invalid email address';
        }

        if (!values.username) {
            errors.username = 'Required';
        }

        if (!values.password) {
            errors.password = 'Required';
        } else if (values.password.length > 20 || values.password.trim().length < 8) {
            errors.password = 'Password must contain characters between 8 - 20';
        }

        return errors;
    };

    handleSubmit = (values) => {
        this.props.signUpRequested(values);
    };

    render() {
        const { error, isSubmitting } = this.props;

        return (
            <Formik
                initialValues={initialValues}
                // eslint-disable-next-line react/jsx-handler-names
                validate={this.handleValidation}
                validateOnBlur={false}
                validateOnChange={false}
                onSubmit={this.handleSubmit}
                render={(props) => {
                    return <SignUp {...props} isSubmitting={isSubmitting} hasError={error} />;
                }}
            />
        );
    }
}

SignUpContainer.propTypes = {
    error: PropTypes.object.isRequired,
    isSubmitting: PropTypes.bool.isRequired,
    signUpRequested: PropTypes.func.isRequired,
};

const mapStateToProps = ({ signUpReducer }) => ({
    error: signUpReducer.error,
    isSubmitting: signUpReducer.isSubmitting,
});

export default connect(mapStateToProps, { signUpRequested })(SignUpContainer);
