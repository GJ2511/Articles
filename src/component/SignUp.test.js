import React from 'react';

import SignUp from './SignUp';

const props = {
    errors: {},
    hasError: {},
    isSubmitting: false,
    touched: {},
};

describe('<SignUp tests />', () => {
    test('match snapshot', () => {
        expect(shallow(<SignUp {...props} />)).toMatchSnapshot();
    });
});
