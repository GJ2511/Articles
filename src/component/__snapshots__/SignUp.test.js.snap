// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp tests /> match snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-9 col-md-7 col-lg-5 mx-auto"
  >
    <div
      className="card card-signUp my-5 bg-light "
    >
      <div
        className="card-header"
      >
        <h5
          className="card-title text-center"
        >
          Sign Up
        </h5>
      </div>
      <div
        className="card-body"
      >
        <Form
          noValidate={true}
        >
          <div
            className="form-group"
          >
            <Field
              aria-describedby="usernameHelp"
              autoFocus={true}
              className="form-control rounded-pill"
              id="username"
              name="username"
              placeholder="User Name"
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <Field
              className="form-control rounded-pill"
              id="email"
              name="email"
              placeholder="Email address"
              type="email"
            />
          </div>
          <div
            className="form-group"
          >
            <Field
              className="form-control rounded-pill"
              id="password"
              name="password"
              placeholder="Password "
              type="password"
            />
          </div>
          <button
            className="btn btn-primary btn-lg btn-block rounded-pill"
            disabled={false}
            type="submit"
          >
            Sign Me Up
          </button>
        </Form>
      </div>
      <div
        className="card-footer"
      >
        Already have an account? 
        <Link
          to="/signin"
        >
           Sign In
        </Link>
      </div>
    </div>
  </div>
</div>
`;
